### PR TITLE
[KYUUBI #2591] Redact secret information from ProcBuilder log

### LINKED ISSUE #2591: [FEATURE] Redact secret information from ProcBuilder log
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the feature

Now the user can see the command to start the engine, which may have some sensitive information.
Introduce a configuration item to support replacing sensitive information.

### Motivation

_No response_

### Describe the solution

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by cxzl25**
### _Why are the changes needed?_
Now the user can see the command to start the engine, which may have some sensitive information.
Introduce a configuration item to support replacing sensitive information.

For example, if you use the `kyuubi.ha.zookeeper.auth.digest` configuration, you can configure `kyuubi.server.redaction.regex`  `(?i)zookeeper.auth.digest`

close #2591

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by cxzl25** [comment]
## Current
image


## Fix
image

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala [hunk 1]
```diff
@@ -185,7 +185,8 @@ private[kyuubi] class EngineRef(
 
     MetricsSystem.tracing(_.incCount(ENGINE_TOTAL))
     try {
-      info(s"Launching engine:\n$builder")
+      val redactedCmd = Utils.redact(conf.get(SERVER_SECRET_REDACTION_PATTERN), builder.toString)
```
Comment (ulysses-you) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala:188]: shall we redact it before combine the submission cmd ？so we can redact the value inside `key=value`.

Comment (cxzl25) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala]: make sense, I try another way.

**Review by ulysses-you — APPROVED** [review]

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 2]
```diff
@@ -1468,4 +1469,12 @@ object KyuubiConf {
       .version("1.6.0")
       .booleanConf
       .createWithDefault(false)
+
+  val SERVER_SECRET_REDACTION_PATTERN: OptionalConfigEntry[Regex] =
+    buildConf("kyuubi.server.redaction.regex")
+      .doc("Regex to decide which Kyuubi contain sensitive information. When this regex matches " +
+        "a property key or value, the value is redacted from the environment UI and various logs.")
```
Comment (ulysses-you) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:1476]: the docs should be refined, only affects the log

**Comment by ulysses-you** [comment]
cc @yaooqinn

**Comment by wForget** [comment]
Can we make `kyuubi.server.redaction.regex` to global config instead of just server ?

**Comment by ulysses-you** [comment]
Engines have their self redcat config, I think it's OK to make this config scope limit to kyuubi server.

**Review by yaooqinn — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master

**Pull request closed without merging**